pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -53,6 +53,15 @@ def on_trainer_init(
             progress_bar_refresh_rate, process_position
         )
 
+    def resolve_checkpoint_path(self):
+        if self.trainer.testing:
+            checkpoint_callbacks = self.trainer.checkpoint_callbacks[0]
+            checkpoint_path = checkpoint_callbacks.best_model_path
```

Comment:
I'd still say this is incorrect. There is no point in doing a reload while testing. What if there is no checkpoint_callback in when doing it a second time? what if I pass a specific ckpt_path, then it will again reload the best_checkpoint_path.

Reply:
If I understand correctly,

For doing inference on a given checkpoint, you can either do:
1. 
```
model = MyLightningModule.from_...(...)
trainer.test(model)
```

2.
```
Trainer(resume_from_checkpoint=...)
trainer.test()
```

which still works !

But now, you can do

3.
```
Trainer(resume_from_checkpoint=...)
trainer.fit() # finetune for some epochs. If new checkpoint is created, this will be the one to be loaded from test
trainer.test()
```

Can you share what is missing and how you think it should be.

Best,
T.C

Reply:
'best' checkpoint is already loaded here before doing any restore:
https://github.com/PyTorchLightning/pytorch-lightning/blob/d1e97a4f114a285349e31e330c7bf8937bc1ee04/pytorch_lightning/trainer/trainer.py#L770-L785

Reply:
and during testing, we don't need to reload anything else like optimizer state/scheduler state .... etc.

Reply:
How will you enable fine-tuning and evaluation of new best model without this change ?

Reply:
```python
# training
trainer = Trainer()
trainer.fit() # saves some checkpoints

# fine-tuning
trainer = Trainer(resume_from_checkpoint=...)
trainer.fit() ## let it reload resume_from_checkpoint and update the checkpoint_callback while fine-tuning.
trainer.test() ## here checkpoint_callback is already updated with the best model after doing fine-tuning, so it will load the best model using the above code, and there is no need to do any restore here.
```